firedrake/interpolation.py

```diff
@@ -543,9 +543,7 @@ def __init__(
         # input ordering VOM will only contain the points on rank 0!
         # QUESTION: Should any of the below have annotation turned off?
         ufl_scalar_element = V_dest.ufl_element()
-        if ufl_scalar_element.num_sub_elements and not isinstance(
-            ufl_scalar_element, finat.ufl.TensorProductElement
-        ):
+        if ufl_scalar_element.num_sub_elements:
```

Comment:
This seems like a bit of a strange condition. Is `num_sub_elements` zero for non-mixed elements? It seems like it would be better for `num_sub_elements` to only be an attribute on mixed elements as it's otherwise kind of oddly defined.